tests/test_asb_save_test_result.yml

```diff
@@ -0,0 +1,65 @@
+---
+- hosts: localhost
+  vars:
+    result_file_path: /var/tmp/test-result
+  roles:
+    - ansibleplaybookbundle.asb-modules
+  tasks:
+    - name: prepare test for clean run
+      file:
+        path: "{{ result_file_path }}"
+        state: absent
+
+    - name: create test result 
+      asb_save_test_result:
+        fail: true
+        msg: "Test failed"
+
+    - stat:
+        path: "{{ result_file_path }}"
+      register: result_file_status
+
+    - name: test result files should exist
+      debug:
+        var: result_file_status.stat.exists
+
+    - fail:
+        msg: Test result file at {{ result_file_path }} could not be found
+      when: not result_file_status.stat.exists
+
+    - name: read credentials file
```

Comment:
You might want to change all the mentions of "credentials" or "encoded" in this file.  Other than that nitpick, the PR looks good to me.